pkg/config/validation/validation.go

```diff
@@ -1254,6 +1256,16 @@ func validateTrafficPolicy(policy *networking.TrafficPolicy) Validation {
 		validatePortTrafficPolicies(policy.PortLevelSettings))
 }
 
+func validateHostForTunnelingPolicy(host string, policy *networking.TrafficPolicy) Validation {
+	if policy == nil || policy.Tunneling == nil {
+		return Validation{}
+	}
+	if strings.Contains(host, "*") {
+		return WrapError(fmt.Errorf("traffic tunneling is not allowed for wildcard domains"))
```

Comment:
Is this reasonable? IIRC the new tcp tunneling approach support downstream SNI as upstream host. 
The SNI value must not be wildcard.

Reply:
By new tcp tunneling approach do you mean this pull request: https://github.com/envoyproxy/envoy/pull/20230? If so, I think that this is reasonable restriction and the only correct for now, because `auto_sni` might be a nice improvement in case of TLS connections, but using it would be much more complicated than I assumed at the beginning, because it wouldn't work for plain TCP connections, so it needs much more effort to handle it properly. What's more, I work on `auto_sni` in the meantime and I don't know at the moment when it will be ready.
Maybe this is not an ideal solution, but it supports both TCP and TLS. We can relax this restriction in the future if it will be really needed.